ui/apps/everest/src/components/confirm-form-dialog/confirm-form-dialog.tsx

```diff
@@ -41,42 +41,51 @@ interface ConfirmFormDialogProps<T extends ConfirmDialogType>
   cancelMessage?: string;
   submitMessage?: string;
   dialogContent?: ReactNode;
+  confirmationInput?: boolean;
+  alertTitle?: string;
+  alertMessage: string;
 }
 export const ConfirmFormDialog = ({
   inputLabel,
   inputPlaceholder,
   handleConfirm,
   cancelMessage = 'Cancel',
   submitMessage = 'Delete',
+  confirmationInput = true,
   selectedId,
   dialogContent,
   submitting,
+  alertTitle = 'Irreversible action',
+  alertMessage = '',
   ...props
 }: ConfirmFormDialogProps<ConfirmDialogType>) => {
   const onSubmit: SubmitHandler<ConfirmDialogType> = (data) => {
     handleConfirm(data);
   };
+
   return (
     <FormDialog
       onSubmit={onSubmit}
-      schema={confirmFormDialogSchema(selectedId)}
+      schema={confirmFormDialogSchema(selectedId, confirmationInput)}
       defaultValues={confirmDialogDefaultValues}
       dataTestId={selectedId && kebabize(selectedId)}
       submitMessage={submitMessage}
       cancelMessage={cancelMessage}
       submitting={submitting}
       {...props}
     >
-      <IrreversibleAction />
+      <IrreversibleAction alertTitle={alertTitle} alertMessage={alertMessage} />
       <DialogContent sx={{ px: 1 }}>{dialogContent}</DialogContent>
-      <TextInput
-        name={ConfirmFormDialogFields.confirmInput}
-        label={inputLabel}
-        labelProps={{ sx: { mt: 0 } }}
-        textFieldProps={{
-          placeholder: inputPlaceholder,
-        }}
-      />
+      {confirmationInput && (
```

Comment:
suggestion: maybe we can create a confirm Form Dialog with different options (like: input/checkbox etc... and each of them will be customizable via boolean and props)? I suppose confirmFormDialog should be universal component and that's why we shouldn't add to this component some meaning sence like "Delete" word in DeleteDataCheckbox, because it can be useful for us in future in some other places where checkbox will be needed not for deletion. What do you think about saving it universal?

Reply:
merged this branch into `EVEREST-959-ui-delete-data-from-backup-storage-when-deleting-a-db` and making the refactoring [here](https://github.com/percona/everest/pull/215)